website/server/controllers/api-v3/user.js

```diff
@@ -1000,6 +1000,10 @@ api.userPurchaseHourglass = {
   url: '/user/purchase-hourglass/:type/:key',
   async handler (req, res) {
     let user = res.locals.user;
+    req.params.hourglass = true;
```

Comment:
Hmm this feels hacky... `req.body.quantity` is lacking validation on it being a number and apidocs and I'd not change the original `req.params` object

Is it necessary for quantity to be in req.params? Cannot you access `req.body.quantity` in `common.ops.buy`?